### PR TITLE
Add global method that allows manipulation of states in Gesture Handler

### PR DESCRIPTION
**Pull request by j-piasecki**
## Description

This PR adds a new `_setGestureState` method to the global object which will be used by the `react-native-gesture-handler`. In order to accomplish synchronous communication between modules I added shared interfaces between libraries (`GestureHandlerStateManager.java` and `RNGestureHandlerStateManager.h`). On startup RNGH module is tried to be loaded and then casted to the correct interface.
I did not update the documentation since this method is intended to be used internally.

## Changes

- Added global method `_setGestureState`
- Added loading of RNGH module and shared interfaces
- Added `Custom` gesture to the relevant Babel plugin list
- Changed `onBegan` to `onBegin` in the Babel plugin

## Checklist

- [ ] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Added TS types tests
- [ ] Added unit / integration tests
- [ ] Updated documentation
- [ ] Ensured that CI passes

### CODE REVIEW

File: src/reanimated2/NativeMethods.ts [hunk 1]
```diff
@@ -38,3 +38,11 @@ export function scrollTo(animatedRef, x, y, animated) {
   const viewTag = animatedRef();
   _scrollTo(viewTag, x, y, animated);
 }
+
+export function setGestureState(handlerTag: number, newState: number): void {
+  'worklet';
+  if (!_WORKLET && !isChromeDebugger()) {
+    return;
```
Comment (kmagiera) [inline src/reanimated2/NativeMethods.ts:45]: Maybe show a warning here?

**Review by kmagiera — APPROVED** [review]
Looks good. Added just one minor comment

File: src/reanimated2/NativeMethods.ts (hunk 1 above)
Comment (piaskowyk) [reply src/reanimated2/NativeMethods.ts:45]:
```suggestion
    console.warn("[Reanimated] You can not use setGestureState in non-worklet function.");
```

**Review by piaskowyk — APPROVED** [review]

File: Common/cpp/Tools/RuntimeDecorator.cpp [hunk 2]
```diff
@@ -207,6 +208,20 @@ void RuntimeDecorator::decorateUIRuntime(
...
       rt, jsi::PropNameID::forAscii(rt, "_stopObservingProgress"), 0, clb8);
   rt.global().setProperty(rt, "_stopObservingProgress", _stopObservingProgress);
+  
+  auto clb9 = [setGestureState](
+                  jsi::Runtime &rt,
+                  const jsi::Value &thisValue,
+                  const jsi::Value *args,
+                  size_t count) -> jsi::Value {
+    int handlerTag = (int)args[0].asNumber();
+    int newState = (int)args[1].asNumber();
```
Comment (piaskowyk) [inline Common/cpp/Tools/RuntimeDecorator.cpp:218]:
```suggestion
    int handlerTag = static_cast<int>(args[0].asNumber());
    int newState = static_cast<int>(args[1].asNumber());
```

**Pull request merged**